src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -229,130 +233,233 @@ public String getUserAnalysisOutputsPage(Model model) {
-		String workflowName = messageSource.getMessage("workflow." + analysisType.getType() + ".title", null, analysisType.getType(), locale);
-		model.addAttribute("workflowName", workflowName);
-		model.addAttribute("version", iridaWorkflow.getWorkflowDescription()
-				.getVersion());
+		String workflowName = messageSource.getMessage("workflow." + analysisType.getType() + ".title", null,
+				analysisType.getType(), locale);
 
-		// Input files
-		// - Paired
-		Set<SequenceFilePair> inputFilePairs = sequencingObjectService.getSequencingObjectsOfTypeForAnalysisSubmission(
-				submission, SequenceFilePair.class);
-		List<SampleFiles> sampleFiles = inputFilePairs.stream().map(SampleFiles::new).sorted((a, b) -> {
-			if (a.sample == null && b.sample == null) {
-				return 0;
-			} else if (a.sample == null) {
-				return -1;
-			} else if (b.sample == null) {
-				return 1;
-			}
-			return a.sample.getLabel()
-					.compareTo(b.sample.getLabel());
-		}).collect(Collectors.toList());
-		model.addAttribute("paired_end", sampleFiles);
+		String version = iridaWorkflow.getWorkflowDescription()
+				.getVersion();
+		String priority = submission.getPriority()
+				.toString();
+		Long duration = getAnalysisDuration(submission);
+
+		AnalysisSubmission.Priority[] priorities = AnalysisSubmission.Priority.values();
+		boolean emailPipelineResult = submission.getEmailPipelineResult();
+
+		boolean canShareToSamples = false;
+		if (submission.getAnalysis() != null) {
+			canShareToSamples = analysisSubmissionSampleProcessor.hasRegisteredAnalysisSampleUpdater(
+					submission.getAnalysis()
+							.getAnalysisType());
+		}
 
 		// Check if user can update analysis
 		Authentication authentication = SecurityContextHolder.getContext()
 				.getAuthentication();
-		model.addAttribute("updatePermission", updateAnalysisPermission.isAllowed(authentication, submission));
+
+		Map<String, Object> detailsPageMap = new HashMap<>();
+		detailsPageMap.put("result", "success");
+		detailsPageMap.put("workflowName", workflowName);
+		detailsPageMap.put("version", version);
+		detailsPageMap.put("priority", priority);
+		detailsPageMap.put("duration", duration);
+		detailsPageMap.put("createdDate", submission.getCreatedDate()
+				.toString());
+		detailsPageMap.put("priorities", priorities);
+		detailsPageMap.put("updatePermission", updateAnalysisPermission.isAllowed(authentication, submission));
+		detailsPageMap.put("canShareToSamples", canShareToSamples);
+		detailsPageMap.put("emailPipelineResult", emailPipelineResult);
+		detailsPageMap.put("updateSamples", submission.getUpdateSamples());
+		return detailsPageMap;
+	}
+
+	/**
+	 * Get analysis input files and their sizes
+	 *
+	 * @param submissionId analysis submission id to get data for
+	 * @return map of input files and their sizes
+	 */
+	@RequestMapping(value = "/ajax/inputs/{submissionId}", method = RequestMethod.GET)
+	public Map<String, Object> ajaxGetAnalysisInputFiles(@PathVariable Long submissionId) {
```

Comment:
Trying to get away from return Maps can you create a dto for this?

Reply:
Fixed in 2804c23baa10fdf170ebd8b07a191142d0927b12